go/libraries/doltcore/sqle/dtables/branches_table.go

```diff
@@ -106,17 +107,24 @@ type BranchItr struct {
 }
 
 // NewBranchItr creates a BranchItr from the current environment.
-func NewBranchItr(ctx *sql.Context, ddb *doltdb.DoltDB, remote bool) (*BranchItr, error) {
+func NewBranchItr(ctx *sql.Context, db dsess.SqlDatabase, remote bool) (*BranchItr, error) {
 	var branchRefs []ref.DoltRef
 	var err error
 
+	txRoot, err := dsess.TransactionRoot(ctx, db)
+	if err != nil {
+		return nil, err
+	}
+
+	ddb := db.DbData().Ddb
+
 	if remote {
-		branchRefs, err = ddb.GetRefsOfType(ctx, map[ref.RefType]struct{}{ref.RemoteRefType: {}})
+		branchRefs, err = ddb.GetRefsOfTypeByNomsRoot(ctx, map[ref.RefType]struct{}{ref.RemoteRefType: {}}, txRoot)
 		if err != nil {
 			return nil, err
 		}
 	} else {
-		branchRefs, err = ddb.GetBranches(ctx)
+		branchRefs, err = ddb.GetBranchesByNomsRoot(ctx, txRoot)
```

Comment:
TODO: This should be getting Refs + Addrs and not re-resolving the refs down below.

Reply:
Will follow up with this